src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,506 @@
+        0 (ground), 1 (excited) and NaN (failed to measure) for all atoms in the result.
+
+        The QuEra results are summarized via 3 values: status, pre_sequence, and post_sequence.
+
+        Status is success or fail. The pre_sequence is 1 if an atom in the ground state was
+        successfully initialized, and 0 otherwise. The post_sequence is 1 if an atom in the
+        ground state was measured, and 0 otherwise. Comparison of pre_sequence and post_sequence
+        reveals one of 4 possible outcomes. The first two (initial measurement of 0) indicate a
+        failure to initialize correctly, and will yeild a NaN result. The second two are measurements
+        of the excited and ground state repsectively, and yield 1 and 0.
+
+        0 --> 0: NaN - Atom failed to be placed (no atom detected in the ground state either before or after)
+        0 --> 1: NaN - Atom failed to be placed (but was recaptured, or something else weird happened)
+        1 --> 0: 1 (Rydberg state) - atom measured in ground state before, but not after
+        1 --> 1: 0 (ground state) - atom measured in ground state both before and after
+        """
+
+        # if entire measurement failed, all NaN
+        if not res.status.value.lower() == "success":
+            return np.array([np.NaN for i in res.pre_sequence])
+
+        # if a single atom failed to initialize, NaN for that individual measurement
+        pre_sequence = [i if i else np.NaN for i in res.pre_sequence]
+
+        # set entry to 0 if ground state is measured, 1 if excited state is measured, NaN if measurement failed
+        return np.array(pre_sequence - res.post_sequence)
+
+
+class BraketAwsAhsDevice(BraketAhsDevice):
+    """Amazon Braket AHS device for hardware in PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers
+            (i.e., ``[-1, 0, 2]``) or strings (``['ancilla', 'q1', 'q2']``).
+        s3_destination_folder (AwsSession.S3DestinationFolder): Name of the S3 bucket
+            and folder, specified as a tuple.
+        shots (int): Number of executions to run to aquire measurements. Defaults to 100.
+        aws_session (Optional[AwsSession]): An AwsSession object created to manage
+            interactions with AWS services, to be supplied if extra control
+            is desired. Default: None
+        poll_timeout_seconds (float): Total time in seconds to wait for
+            results before timing out.
+        poll_interval_seconds (float): The polling interval for results in seconds.
+
+    """
+
+    name = "Braket Device for AHS in PennyLane"
+    short_name = "braket.aws.ahs"
+
+    def __init__(
+            self,
+            wires: Union[int, Iterable],
+            device_arn: str,
+            s3_destination_folder: AwsSession.S3DestinationFolder = None,
+            *,
+            shots: int = 100,
```

Comment:
Should the default number of shots be set by the backend? e.g., `AwsDevice.DEFAULT_SHOTS_QPU` (same question for the simulator version)